src/codestyle.sh

```diff
@@ -0,0 +1,161 @@
+# Checkpatch is a useful tool provided by Linux, and the main goal of the code
+# in this file is to handle this script in a way to make this tool easier for
+# users.
+
+include "${KW_LIB_DIR}/lib/kw_config_loader.sh"
+include "${KW_LIB_DIR}/lib/kwlib.sh"
+
+declare -gA options_values
+
+# Runs checkpatch in the given path, which might be a file or directory.
+#
+# @FILE_OR_DIR_CHECK Target path for running checkpatch script
+function codestyle_main()
+{
+  local path
+  local flag
+  local checkpatch_options="${configurations[checkpatch_opts]}"
+  local -r original_working_dir="$PWD"
+  local kernel_root
+  local checkpatch
+  local cmd_script
+
+  parse_codestyle_options "$@"
+  if [[ "$?" != 0 ]]; then
+    complain "Invalid option: ${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  flag=${options_values['TEST_MODE']}
+  [[ -n "${options_values['VERBOSE']}" ]] && flag='VERBOSE'
+
+  flag=${flag:-'SILENT'}
+  # TODO: Note that codespell file is not specified yet because of the poluted
+  # output. It could be nice if we can add another option just for this sort
+  # of check.
+
+  path="${options_values['PATH']}"
+  path=${path:-'.'}
+  validate_path "$path"
```

Comment:
As the check is really simple and lean, I think we can remove `validate_path` and explicitly do the check here.